design/healcheck-endpoint.md

```diff
@@ -0,0 +1,47 @@
+# Healthcheck endpoint
+
+* **Status**: Draft #1
+* **JIRA**: [KEYCLOAK-12398](https://issues.redhat.com/browse/KEYCLOAK-12398)
+
+## Motivation
+
+Deploying both Keycloak and RHSSO on top of Kubernetes requires defining both Readiness Check and Liveness Check (see the [manual](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/)). Currently for Keycloak we use a very naive implementation, that calls `/auth/realms/master` URL. This approach has many drawbacks, here are a few examples:
+
+* We can not discover database failures
+* We can not discover if certain Identity Provider is up and running
+* Undertow subsystem is down when proceeding with migration
+
+In order to address this problems, we should provide a parametrized Healthcheck endpoint in the Server and use it in our container image as well as in the Operator. 
+
+## Parametrizing the Healthcheck URL
+
+The preferable approach is to hook into Health Endpoint provided by Wildfly (starting from 16) at:
```

Comment:
Have you checked if it is possible to add query parameters to the WF health endpoint?

Reply:
Over the time, there's one more option I need to investigate - Microprofile Health. It's part of both Wildfly and Quarkus. I'll dive into it shortly.